### PR TITLE
get host address from driver socket

### PR DESCRIPTION
**Pull request by flybywind**
try to fix the problem decribed in issue 397: 
https://github.com/Azure/mmlspark/issues/397

### CODE REVIEW

**Comment by flybywind** [comment]
thanks, and I have tested it on a large dataset in our spark, results seemed well

File: tools/runme/build.sh [hunk 1]
```diff
@@ -109,13 +109,17 @@ _sbt_build() {
...
     if [[ "$TESTS" = "" ]]; then TESTS="all"; fi
   fi
-  _sbt_run "full-build"
-  show section "Sorting assembly jar for the maven repo"
-  # leave only the -assembley jars under the proper name (and the pom files)
-  local f; for f in "$BUILD_ARTIFACTS/packages/m2/"**; do case "$f" in
-    ( *-@(javadoc|sources).jar@(|.md5|.sha1) ) _rm "$f" ;;
-    ( *-assembly.jar@(|.md5|.sha1) ) _ mv "$f" "${f//-assembly.jar/.jar}" ;;
-  esac; done
+  if [[ "$SUBPROJECT" != "" ]]; then
```
Comment (imatiach-msft) [inline tools/runme/build.sh]: @flybywind sorry, one question, why was this change necessary in build.sh?

Comment (imatiach-msft) [reply tools/runme/build.sh]: @flybywind I'm not sure if this specific change is necessary in this file, can it be removed?

File: tools/runme/build.sh [hunk 2]
```diff
@@ -109,13 +109,17 @@ _sbt_build() {
...
   fi
-  _sbt_run "full-build"
-  show section "Sorting assembly jar for the maven repo"
-  # leave only the -assembley jars under the proper name (and the pom files)
-  local f; for f in "$BUILD_ARTIFACTS/packages/m2/"**; do case "$f" in
-    ( *-@(javadoc|sources).jar@(|.md5|.sha1) ) _rm "$f" ;;
-    ( *-assembly.jar@(|.md5|.sha1) ) _ mv "$f" "${f//-assembly.jar/.jar}" ;;
-  esac; done
+  if [[ "$SUBPROJECT" != "" ]]; then 
+      _sbt_run "project $SUBPROJECT" compile
```
Comment (mhamilton723) [inline tools/runme/build.sh]: Can we revert this change? Ill take care of cleaning up the build in a separate large PR

Comment (flybywind) [reply tools/runme/build.sh]: when i ran `sbt 'project lightbm' compile`, it appeared than many variables was missing. 
After reviewing the build.sh I saw it will make some exporting of the variables to the env,
so I changed the build.sh.
I will revert the changes and you do the fixing, thank you !

File: tools/runme/build.sh (hunk 1 above)
Comment (flybywind) [reply tools/runme/build.sh]: i have reverted the change and leave the problem to @mhamilton723

**Comment by imatiach-msft** [comment]
I verified this PR on a spark cluster with the build, @mhamilton723 could you please approve the PR? Thanks!
